### PR TITLE
pages: default page template

### PR DESCRIPTION
**Pull request by jalavik**
* Sets default template and avoids crashing when no template
  is specified in admin interface.

Reviewed-by: Jan Aage Lavik <jan.age.lavik@cern.ch>
Signed-off-by: Javier Martin Montull <javier.martin.montull@cern.ch>

### CODE REVIEW

File: invenio/modules/pages/admin.py [hunk 1]
```diff
@@ -60,6 +62,7 @@ class PagesAdmin(ModelView):
 
     form_args = dict(
         template_name=dict(
+            default="default.html",
```
Comment (bouzlibop) [inline invenio/modules/pages/admin.py]: Shouldn't we go here with sth like `PAGES_DEFAULT_TEMPLATE` from `pages/config.py`

File: invenio/modules/pages/admin.py [hunk 2]
```diff
@@ -33,6 +33,8 @@
 
 def template_exists(form, field):
     """Form validation: check that selected template exists."""
+    if not field.data:
```
Comment (jirikuncar) [inline invenio/modules/pages/admin.py]: what about adding 'required' validator to https://github.com/inveniosoftware/invenio/pull/3337/files#diff-b83bf3bae0fb93452518c475004ffe97R66 instead?

File: invenio/modules/pages/admin.py (hunk 1 above)
Comment (jmartinm) [inline invenio/modules/pages/admin.py]: I was looking at adding:

```python
default=os.path.basename(cfg["PAGES_DEFAULT_TEMPLATE"])
```

here but then I get:

```
RuntimeError: working outside of application context
```

How can the app config be accessed from the class definition?

File: invenio/modules/pages/admin.py (hunk 2 above)
Comment (jmartinm) [inline invenio/modules/pages/admin.py]: Makes sense. Will add that.

File: invenio/modules/pages/admin.py [hunk 3]
```diff
@@ -33,6 +33,8 @@
...
+        return
     template_name = "pages/" + field.data
```
Comment (jirikuncar) [inline invenio/modules/pages/admin.py]: consider using `werkzeug. secure_filename`

File: invenio/modules/pages/admin.py (hunk 1 above)
Comment (bouzlibop) [inline invenio/modules/pages/admin.py]: Hmm, I'm not sure how to do it. @jirikuncar, do you know how could this be solved?

Comment (jirikuncar) [inline invenio/modules/pages/admin.py]: It used to work with lambda function or alternatively you can wrap it to a LocalProxy.

**Comment by jmartinm** [comment]
I will create a PR with these changes for `Invenio-pages`

File: invenio/modules/pages/admin.py [hunk 4]
```diff
@@ -60,7 +69,8 @@ class PagesAdmin(ModelView):
 
     form_args = dict(
         template_name=dict(
-            validators=[template_exists]
+            default=LocalProxy(lambda: os.path.basename(cfg["PAGES_DEFAULT_TEMPLATE"])),
```
Comment (jirikuncar) [inline invenio/modules/pages/admin.py]: btw have you checked if it works only with lambda function?

**Comment by jmartinm** [comment]
superseded by https://github.com/inveniosoftware/invenio-pages/pull/3

File: invenio/modules/pages/admin.py (hunk 4 above)
Comment (jmartinm) [inline invenio/modules/pages/admin.py]: Yes, it seems to work with:

```python
default=lambda: os.path.basename(cfg["PAGES_DEFAULT_TEMPLATE"]),
```

Comment (jmartinm) [inline invenio/modules/pages/admin.py]: I updated inveniosoftware/invenio-pages#3

**Pull request closed without merging**